### ISSUE TITLE
Assetic package 'ezdesign' not working in a {% stylesheets %} block

### ISSUE THREAD

**Issue by alafon** [lolautruche/EzCoreExtraBundle#10]
This works :

```
<link rel="stylesheet" href="{{ asset("css/default.css", "ezdesign") }}" />
<link rel="stylesheet" href="{{ asset("css/another_sheet.css", "ezdesign") }}" />
```

This does not :

```
{% stylesheets
    'css/default.css'
    'css/another_sheet.css'
    package='ezdesign'
%}
    <link rel="stylesheet" type="text/css" href="{{ asset_url }}"/>
{% endstylesheets %}

```

Am I doing something wrong ?

**Comment by lolautruche** [comment]
Never tried this... Didn't know about `package` option.
Do you have details on the error?

**Comment by lolautruche** [comment]
Anyway, I doubt this can work at all since Assetic compilation doesn't happen at run time, so you'll lose siteaccess context.

Best would be either to define [assetic name assets instead](https://symfony.com/doc/current/assetic/asset_management.html#using-named-assets), or just use Grunt/Bower

**Comment by alafon** [comment]
The `package` option is in the [Asset Component](http://symfony.com/blog/new-in-symfony-2-7-the-new-asset-component#template-function-changes), so it has nothing to do with assetic, you're right, sorry.

I don't know assetic very well but maybe custom filters could be used to lookup for the assets within the themes.

I was not able to find a lot of documentation regarding custom filter development and reverse engineering the source code of assetic is... well... 🗡 .
Found this at least :
- http://richardmiller.co.uk/2011/05/24/symfony2-make-your-own-assetic-filter/
- http://stackoverflow.com/questions/8136177/how-to-create-own-symfony-assetic-filters

If you want to add some hints and finally think that it can be accomplished I might give a hand 😄

**Comment by lolautruche** [comment]
> I don't know assetic very well but maybe custom filters could be used to lookup for the assets within the themes.

Yes, but as I said, Assetic will know nothing about your siteaccess config. When you run `assetic:dump`, you'll use default siteaccess, so it's useless when using themes with different siteaccesses...

That's why I suggest to use named assets.

Assetic is quite rigid as it can't use dynamic references (e.g. variables inside `stylesheets`/`javascripts` tags)
